namespaces/exec.go

```diff
@@ -110,9 +110,36 @@ func Exec(container *libcontainer.Config, stdin io.Reader, stdout, stderr io.Wri
 			return -1, err
 		}
 	}
+	if !container.Namespaces.Contains(libcontainer.NEWPID) {
+		killAllPids(container)
+	}
 	return command.ProcessState.Sys().(syscall.WaitStatus).ExitStatus(), nil
 }
 
+func killAllPids(container *libcontainer.Config) {
+	if systemd.UseSystemd() {
+		systemd.Freeze(container.Cgroups, cgroups.Frozen)
+		if pids, err := systemd.GetPids(container.Cgroups); err == nil {
+			for _, pid := range pids {
+				if p, err := os.FindProcess(pid); err == nil {
+					p.Kill()
```

Comment:
Do you think we should also do a wait4 after we send the kill signal to each process?